script/client

```diff
@@ -181,6 +181,7 @@ if ($options{form} && $options{'json-data'}) {
     exit(2);
 }
 
-exit OpenQA::Client::run(\%options, @ARGV);
+my $ret = run(\%options, @ARGV);
+exit(ref $ret eq 'HASH' ? 0 : 1);
```

Comment:
I don't think we have strict requirement that all results must contain json - sometimes status code is enough, so this check will not be reliable?

Reply:
and theoretically, `$json` could be an arrayref also.

Why would `run/handle_result` return the json at all? Can't it be changed to return 0?
This line looks like a bad hack :)

Reply:
Not sure what you mean. This is basically the return code from `handle_result` which can either be the response from the remote server which is always a hash or it's a literal "1".

Reply:
@perlpunk I can try to change this to not return `$json` at all then

Reply:
done

Reply:
Hm, I just saw that there are still functions calling `handle_result` and expecting data.
I think it was working before in `script/client` because it just did an `exit` in case of an error.
A cleaner way would probably be to throw an exception in case of an error.
But this would require more changes.
Maybe put the `exit` back at the end of `handle_result`...

Reply:
@okurz regarding the result not being a hash - AFAIK there are some endpoints not returning JSON at all, like @andrii-suse said. Just don't know which right now.

Reply:
That would make testing harder as we would need to catch the exit from a process rather than just check returns. Should I use `ref $ret eq 'HASH'` or rather `ref $ret eq 'SCALAR' ? 1 : 0`?

Reply:
https://github.com/os-autoinst/openQA/blob/master/lib/OpenQA/WebAPI/Controller/API/V1/Locks.pm#L65